aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/ExecutorStepTest.java

```diff
@@ -414,5 +420,77 @@ private void startJnlpProc() throws Exception {
             }
         });
     }
+    
+    /**
+     * Ensures that {@link FlowListener} callback is being invoked for the {@link ExecutorStepExecution}.
+     * It allows to record the executor, which is being utilized by the flow.
+     */
+    @Test
+    @Issue("JENKINS-32225")
+    public void shouldInvokeFlowListenerCallbacksAndRecordNodes() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                DumbSlave s = createSlave(story.j);
+                s.setLabelString("remote quick");
+                s.getNodeProperties().add(new EnvironmentVariablesNodeProperty(
+                        new EnvironmentVariablesNodeProperty.Entry("ONSLAVE", "true")));
+
+                // Create a project invoking the node() operation to acquire an executor
+                p = jenkins().createProject(WorkflowJob.class, "demo");
+                p.setDefinition(new CpsFlowDefinition(
+                    "node('" + s.getNodeName() + "') {\n" +
+                    "    echo 'Hello'\n" +
+                    "}"));
+
+                // Run build and wait for its completion
+                story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
 
+                // Get the node invocation monitor
+                NodeInvocationListener listener = story.j.jenkins.getExtensionList(FlowListener.class)
+                        .get(NodeInvocationListener.class);
+                assertThat("NodeInvocationListener must be specified in the test", listener, notNullValue());
+                Node builtOn = listener.getBuiltOn();
+                assertThat("Node has not been recorded", builtOn, notNullValue());
```

Comment:
Using a negative sentence seems confusing, "Node has to be recorded" is better IMO so it reads "Assert that: node has to be recorded".